components/email_aliases/email_aliases.mojom

```diff
@@ -0,0 +1,62 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+module email_aliases.mojom;
+
+struct Alias {
+  string email;
+  string? note;
+  array<string>? domains;
+};
+
+// Observer for change in email aliases state
+interface EmailAliasesServiceObserver {
+
+  // Called when aliases have been updated.
+  OnAliasesUpdated(array<Alias> aliases);
+
+  // Called when user logs in to an account.
+  OnLoggedIn(string primaryEmail);
+
+  // Called when user has logged out of account.
+  OnLoggedOut();
+
+  // Called when account verification is pending.
+  OnVerificationPending(string pendingAccount);
```

Comment:
how do we know when it finishes? Also how do we know if verification has already been completed in some previous session?

Reply:
Why would observers be triggered when the UI is launched? That seems a bit unconventional, are we doing this in other UIs? Either way my question is about how the UI knows that it finished. I don't see any notification for when the verification is complete?

Reply:
Hey, I think you could easily handle the `PendingVerification` state if you had a `pending_verification` boolean on your Alias struct:
https://github.com/brave/brave-core/pull/28543/files#r2085517249

Reply:
@bridiver Verification is complete when OnLoggedIn is called. The flow is as follows:

1. Logged out
2. `RequestPrimaryEmailVerification()`
3. `OnVerificationPending()`
4. (user clicks the link in the verification email, outside the UI)
5. `OnLoggedIn`
6. OnAliasesUpdated --> List of aliases on this account displayed

Reply:
@fallaciousreasoning To clarify: `PendingVerification` is unrelated to Aliases. Instead it's talking about account verification using the user's primary email address (not an Alias address).

Reply:
I hope I have clarified what is happening via https://github.com/brave/brave-core/pull/28543#discussion_r2089634549